pkg/storaged/lvm2/create-logical-volume-dialog.jsx

```diff
@@ -0,0 +1,298 @@
+                                   tooltip: _("Save space by storing identical data blocks just once")
+                               },
+                           ],
+                           value: {
+                               compression: true,
+                               deduplication: true,
+                           }
+                       }),
+        ],
+        update: (dlg, vals, trigger) => {
+            if (vals.purpose == 'block' && (trigger == "layout" || trigger == "pvs" || trigger == "purpose")) {
+                for (const lay of layouts) {
+                    if (lay.value == vals.layout) {
+                        dlg.set_options("pvs", {
+                            min_selected: lay.min_pvs,
+                            explanation: min_pvs_explanation(vals.pvs, lay.min_pvs)
+                        });
+                    }
+                }
+                dlg.set_options("layout",
+                                {
+                                    choices: compute_layout_choices(vals.pvs),
+                                    explanation: layout_descriptions[vals.layout]
+                                });
+                const max = max_size(vals);
+                const old_max = dlg.get_options("size").max;
+                if (vals.size > max || vals.size == old_max)
+                    dlg.set_values({ size: max });
+                dlg.set_options("size", { max });
+            } else if (trigger == "purpose") {
+                dlg.set_options("size", { max: vgroup.FreeSize });
+            }
+        },
+        Action: {
+            Title: _("Create"),
+            action: (vals, progress) => {
+                if (vals.purpose == "block") {
+                    if (!can_do_layouts)
+                        return vgroup.CreatePlainVolume(vals.name, vals.size, { });
+                    else {
+                        return vgroup.CreatePlainVolumeWithLayout(vals.name, vals.size, vals.layout,
+                                                                  vals.pvs.map(spc => spc.block.path),
+                                                                  { });
+                    }
+                } else if (vals.purpose == "pool")
+                    return vgroup.CreateThinPoolVolume(vals.name, vals.size, { });
+                else if (vals.purpose == "vdo") {
+                    return (need_vdo_install ? install_package(vdo_package, progress) : Promise.resolve())
+                            .then(() => {
+                                progress(_("Creating VDO device")); // not cancellable any more
+                                return vgroup.CreateVDOVolume(
+                                // HACK: emulate lvcreate's automatic pool name creation until
+                                // https://github.com/storaged-project/udisks/issues/939
+                                    vals.name, next_default_logical_volume_name(client, vgroup, "vpool"),
```

Comment:
This was actually fixed last year, but let's not drop it here, but in a follow-up.